test/components/preprocessors/test_recursive_splitter.py

```diff
@@ -0,0 +1,258 @@
+    )  # noqa: E501
+
+
+def test_recursive_splitter_empty_documents():
+    splitter = RecursiveDocumentSplitter(split_length=20, split_overlap=0, separators=["."])
+    empty_doc = Document(content="")
+    doc_chunks = splitter.run([empty_doc])
+    doc_chunks = doc_chunks["documents"]
+    assert len(doc_chunks) == 0
+
+
+def test_recursive_chunker_with_multiple_separators_recursive():
+    splitter = RecursiveDocumentSplitter(
+        split_length=260, split_overlap=0, separators=["\n\n", "\n", ".", " "], keep_separator=True
+    )
+    text = """Artificial intelligence (AI) - Introduction
+
+AI, in its broadest sense, is intelligence exhibited by machines, particularly computer systems.
+AI technology is widely used throughout industry, government, and science. Some high-profile applications include advanced web search engines; recommendation systems; interacting via human speech; autonomous vehicles; generative and creative tools; and superhuman play and analysis in strategy games."""  # noqa: E501
+
+    doc = Document(content=text)
+    doc_chunks = splitter.run([doc])
+    doc_chunks = doc_chunks["documents"]
+
+    assert len(doc_chunks) == 4
+    assert (
+        doc_chunks[0].meta["original_id"]
+        == doc_chunks[1].meta["original_id"]
+        == doc_chunks[2].meta["original_id"]
+        == doc_chunks[3].meta["original_id"]
+        == doc.id
+    )
+    assert doc_chunks[0].content == "Artificial intelligence (AI) - Introduction\n\n"
+    assert (
+        doc_chunks[1].content
+        == "AI, in its broadest sense, is intelligence exhibited by machines, particularly computer systems.\n"
+    )
+    assert doc_chunks[2].content == "AI technology is widely used throughout industry, government, and science."
+    assert (
+        doc_chunks[3].content
+        == " Some high-profile applications include advanced web search engines; recommendation systems; interacting via human speech; autonomous vehicles; generative and creative tools; and superhuman play and analysis in strategy games."
+    )
+
+
+def test_recursive_chunker_split_document_with_overlap():
+    splitter = RecursiveDocumentSplitter(split_length=20, split_overlap=11, separators=[".", " "], keep_separator=True)
+    text = """A simple sentence1. A bright sentence2. A clever sentence3. A joyful sentence4"""
+
+    doc = Document(content=text)
+    doc_chunks = splitter.run([doc])
+    doc_chunks = doc_chunks["documents"]
+
+    assert len(doc_chunks) == 4
+    assert (
+        doc_chunks[0].meta["original_id"]
+        == doc_chunks[1].meta["original_id"]
+        == doc_chunks[2].meta["original_id"]
+        == doc_chunks[3].meta["original_id"]
+        == doc.id
+    )
+
+    assert doc_chunks[0].content == "A simple sentence1."
+    assert doc_chunks[0].meta["split_id"] == 0
+    assert doc_chunks[0].meta["split_idx_start"] == 0
```

Comment:
I'd prefer to use this style of checking since this way we can be 100% confident that the `split_idx_start` is correct with respect to the original text. Could we update all `split_idx_start` checks to this style? It's the same one we use for the tests for the NLTKDocumentSplitter. 
```suggestion
    assert doc_chunks[0].meta["split_idx_start"] == text.index(doc_chunks[0].content)
```

Reply:
good point 👍🏽